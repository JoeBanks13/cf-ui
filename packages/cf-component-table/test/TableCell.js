import React from 'react';
import renderer from 'react-test-renderer';
import { TableCell, createTableCell } from '../../cf-component-table/src/index';
import { felaSnapshot } from 'cf-style-provider';

describe('TableCell', () => {
  it('should render', () => {
    const snapshot = felaSnapshot(<TableCell>TableCell</TableCell>);
    expect(snapshot).toMatchSnapshot();
  });

  it('should render with align', () => {
    const snapshot = felaSnapshot(
      <TableCell align="center">TableCell</TableCell>
    );
    expect(snapshot).toMatchSnapshot();
  });

  it('should compose with styles overrides', () => {
    const TableCell = createTableCell(({ theme }) => ({
      verticalAlign: 'right',
      [`@media (min-width: ${theme.breakpoints.desktopLarge})`]: {
        width: '1000px'
      }
    }));
    const snapshot = felaSnapshot(<TableCell>TableCell</TableCell>);
    expect(snapshot).toMatchSnapshot();
  });
});
