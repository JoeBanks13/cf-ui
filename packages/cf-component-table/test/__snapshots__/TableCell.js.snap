// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TableCell should compose with styles overrides 1`] = `
Object {
  "component": <td
    className="19q41br"
    colSpan={undefined}
    rowSpan={undefined}
>
    TableCell
</td>,
  "styles": ".19q41br {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: right;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-color: #dedede
}@media (max-width: 47.2em){.19q41br {
  display: block;
  width: 100%;
  box-sizing: border-box;
  border-top-width: 0px
}}@media screen and (max-width: 47.2em){.19q41br {
  display: block;
  border: none;
  float: left;
  clear: left;
  padding: 0px
}}@media (min-width: 97.6em){.19q41br {
  width: 1000px
}}",
}
`;

exports[`TableCell should render 1`] = `
Object {
  "component": <td
    className="c5609r"
    colSpan={undefined}
    rowSpan={undefined}
>
    TableCell
</td>,
  "styles": "
.c5609r {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: middle;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-color: #dedede
}@media (max-width: 47.2em){
    .c5609r {
  display: block;
  width: 100%;
  box-sizing: border-box;
  border-top-width: 0px
}}@media screen and (max-width: 47.2em){
    .c5609r {
  display: block;
  border: none;
  float: left;
  clear: left;
  padding: 0px
}}",
}
`;

exports[`TableCell should render with align 1`] = `
Object {
  "component": <td
    className="2n3suq"
    colSpan={undefined}
    rowSpan={undefined}
>
    TableCell
</td>,
  "styles": ".2n3suq {
  border-top: 1px solid #dedede;
  line-height: 1.5;
  padding: 0.73333333333rem;
  vertical-align: middle;
  text-align: center;
  white-space: nowrap;
  overflow: hidden;
  text-overflow: ellipsis;
  border-color: #dedede
}@media (max-width: 47.2em){.2n3suq {
  display: block;
  width: 100%;
  box-sizing: border-box;
  border-top-width: 0px
}}@media screen and (max-width: 47.2em){.2n3suq {
  display: block;
  border: none;
  float: left;
  clear: left;
  padding: 0px
}}",
}
`;
